### ISSUE TITLE
Empty metadata value leads to bean-check traceback

### ISSUE THREAD

**Issue by blais** [beancount/beancount#378]
**[Original report](https://bitbucket.org/blais/beancount/issue/378) by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

```
Traceback (most recent call last):
  File "/home/tbm/.local/bin/bean-check", line 6, in <module>
    sys.exit(main())
  File "/home/tbm/.local/lib/python3.7/site-packages/beancount/scripts/check.py", line 40, in main
    extra_validations=validation.HARDCORE_VALIDATIONS)
  File "/home/tbm/.local/lib/python3.7/site-packages/beancount/loader.py", line 90, in load_file
    _log_errors(errors, log_errors)
  File "/home/tbm/.local/lib/python3.7/site-packages/beancount/loader.py", line 128, in _log_errors
    printer.print_errors(errors, file=log_errors)
  File "/home/tbm/.local/lib/python3.7/site-packages/beancount/parser/printer.py", line 444, in print_errors
    output.write(format_error(error))
  File "/home/tbm/.local/lib/python3.7/site-packages/beancount/parser/printer.py", line 414, in format_error
    error_string = '\n'.join(format_entry(entry) for entry in entries)
  File "/home/tbm/.local/lib/python3.7/site-packages/beancount/parser/printer.py", line 414, in <genexpr>
    error_string = '\n'.join(format_entry(entry) for entry in entries)
  File "/home/tbm/.local/lib/python3.7/site-packages/beancount/parser/printer.py", line 339, in format_entry
    return EntryPrinter(dcontext, render_weights)(entry)
  File "/home/tbm/.local/lib/python3.7/site-packages/beancount/parser/printer.py", line 117, in __call__
    method(obj, oss)
  File "/home/tbm/.local/lib/python3.7/site-packages/beancount/parser/printer.py", line 203, in Transaction
    self.write_metadata(posting.meta, oss, '    ')
  File "/home/tbm/.local/lib/python3.7/site-packages/beancount/parser/printer.py", line 143, in write_metadata
    raise ValueError("Unexpected value: '{!r}'".format(value))
ValueError: Unexpected value: 'None'
```

Input file:

```
;2019-01-01 open Assets:A
;2019-01-01 open Assets:B

2019-02-28 txn "Test"
  Assets:A                       10.00 USD
  Assets:B                      -10.00 USD
  foo:
```

(The accounts must not be declared; if they are declared, it works)

Command:

    bean-check d

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Oh that's a bug indeed.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

FWIW, the issue here was in the printer, not in the parser. The parser already translates this missing value as None, despite the fact that it does not accept a "NULL" token to do that explicitly.

Okay, so I'm going to allow parsing the special value NULL in metadata, and render it as well, so this properly fixes the inconsistencies.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #378: Rendering metadata with null values wasn't properly implemented.
